scrapy_zyte_api/providers.py

```diff
@@ -112,10 +125,36 @@ async def __call__(  # noqa: C901
                     options["extractFrom"] = extract_from.value
                     break
 
+        http_response_needed = (
+            AnyResponse in to_provide
+            and BrowserResponse not in to_provide
+            and BrowserHtml not in to_provide
+            and not http_response
+        )
+
+        extract_from = None  # type: ignore[assignment]
         for item_type, kw in item_keywords.items():
             options_name = f"{kw}Options"
             if item_type not in to_provide_stripped and options_name in zyte_api_meta:
                 del zyte_api_meta[options_name]
+            elif options_name in zyte_api_meta:
+                extract_from = zyte_api_meta[options_name].get("extractFrom")
```

Comment:
I think currently it's expected to work fine, but if we add any other option to the API, then extract_from could become None, and it seems this case is not handled later. 

Would it make sense to change the logic so that the condition below is still executed if extractFrom is None after this line?

Reply:
Good call. Updated in https://github.com/scrapy-plugins/scrapy-zyte-api/pull/161/commits/3fec1b8fa5e06e876da14ac5a494970bd8b8bcf3 and added tests for this